src/main/cpp/blaze.cc

```diff
@@ -1426,10 +1426,24 @@ static map<string, EnvVarValue> PrepareEnvironmentForJvm() {
 
     // Disable HTTP proxies for localhost and any localhost-like address,
     // in case we (or gRPC, etc.) ever use one of these addresses.
-    std::string localhost_addresses = "localhost,127.0.0.1,0:0:0:0:0:0:0:1,::1";
+    string localhost_addresses = blaze::GetEnv("no_proxy");
+    vector<string> localhost_addresses_vec{"localhost", "127.0.0.1", "0:0:0:0:0:0:0:1", "::1"};
```

Comment:
This should be below the next block. Setting localhost_addresses is one logical unit.